spot_rl_experiments/spot_rl/skills/atomic_skills.py

```diff
@@ -260,144 +437,244 @@ def __init__(self, spot, config, use_mobile_pick=False):
         """
-        Resets the env and policy
+        Sanity check for the input goal_dict
+
+        Args:
+            goal_dict: Dict[str, Any] containing necessary goal information for picking
+                                      Required key: "target_object" (x,y,theta) where robot needs to navigate to
+                                                    "take_user_input" (bool) Whether to take user input for the success of the gaze
+
+        Raises:
+            KeyError: If the required keys are not found in the goal_dict
+        """
+        target_obj_name = goal_dict.get("target_object", None)  # type: str
+        take_user_input = goal_dict.get("take_user_input", False)  # type: bool
+        if target_obj_name is None:
+            raise KeyError(
+                "Error in Pick.sanity_check(): target_object key not found in goal_dict"
+            )
+
+        conditional_print(
+            message=f"SanityCheck passed, now picking {target_obj_name}, {'WILL' if take_user_input else 'WILL NOT'} take user input at the end",
+            verbose=self.verbose,
+        )
+
+    def reset_skill(self, goal_dict: Dict[str, Any]) -> Any:
+        """
+        Performs:
+            Sanity checks the input goal_dict,
+            Resets the env and policy,
+            Updates the logged data at init,
+            Sets the start time for recording before starting gaze
 
         Args:
-            target_obj_name (str): Name of the target object
+            goal_dict: Dict[str, Any] containing necessary goal information for picking
+                                      Required key: "target_object" (x,y,theta) where robot needs to navigate to
+                                                    "take_user_input" (bool) Whether to take user input for the success of the gaze
 
         Returns:
             observations: observations from the env
 
         """
-        observations = self.env.reset(target_obj_name=target_obj_name)
+        try:
+            self.sanity_check(goal_dict)
+        except Exception as e:
+            raise e
+
+        target_obj_name = goal_dict.get("target_object", None)
+        # Reset the env and policy
+        observations = self.env.reset(
+            target_obj_name=goal_dict["target_object"]
+        )  # safe to access as sanity check passed
         self.policy.reset()
 
+        # Logging and Debug
+        self.env.say(f"Gaze at target object - {target_obj_name}")
+        # Update logged data at init
+        # Dict to store result of pick execution (this also resets the previous data)
+        self.gaze_result = {}
+        # Set start time for recording before starting gaze (this also resets the previous start time data)
+        self.start_time = time.time()
+
         return observations
 
-    def execute_pick(self, target_object_list, take_user_input=False):
+    def execute(self, goal_dict: Dict[str, Any]) -> Tuple[bool, str]:  # noqa
```

Comment:
This was done in such a manner because `execute` had to be kept generalized for atomic & composite skills.

All atomic skills have `execute` behavior in one way but composite skill would have different way.

But for now, moved to base class for Nav & Pick. Keeping in derived class for Place because place.execute() also has extra code for handling IK based arm motion.